src/VS.Mefx/Commands/MatchChecker.cs

```diff
@@ -0,0 +1,446 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace VS.Mefx.Commands
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Globalization;
+    using System.Linq;
+    using Microsoft;
+    using Microsoft.VisualStudio.Composition;
+
+    /// <summary>
+    /// Method to perform matching between parts.
+    /// </summary>
+    internal class MatchChecker : Command
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MatchChecker"/> class.
+        /// </summary>
+        /// <param name="derivedInfo">ConfigCreator for the specified input parts.</param>
+        /// <param name="arguments">Input arguments specified by the user.</param>
+        public MatchChecker(ConfigCreator derivedInfo, CLIOptions arguments)
+            : base(derivedInfo, arguments)
+        {
+        }
+
+        /// <summary>
+        /// Method to perform matching on the input options and output the result to the user.
+        /// </summary>
+        public void PerformMatching()
+        {
+            if (this.Options.MatchParts == null || this.Options.MatchParts.Count == 0)
+            {
+                return;
+            }
+
+            if (this.Options.MatchParts.Count() == 2)
+            {
+                string exportPartName = this.Options.MatchParts.ElementAt(0).Trim();
```

Comment:
Why use the `ElementAt` method when you can use the indexer (`this.Options.MatchParts[0].Trim()`)?